Fix flaky tests in TestChildProcessCleanup (test_stdio.py)

### Initial Checks

- [x] I confirm that I'm using the latest version of MCP Python SDK
- [x] I confirm that I searched for my issue in https://github.com/modelcontextprotocol/python-sdk/issues before opening this issue

### Description

Three tests in `TestChildProcessCleanup` are flaky. They fail intermittently on macOS (darwin) with the same root cause: the child process hasn't started writing to the marker file before the assertion checks.

**Affected tests:**
- `TestChildProcessCleanup.test_basic_child_process_cleanup`
- `TestChildProcessCleanup.test_nested_process_tree`
- `TestChildProcessCleanup.test_early_parent_exit`

**CI evidence:** Tests observed failing on 2025-11-12 and continue to be flaky.

**Output:**

```python
________ TestChildProcessCleanup.test_basic_child_process_cleanup ________
[gw9] darwin -- Python 3.11.14

    async def test_basic_child_process_cleanup(self):
        ...
        # Verify child is writing
        if os.path.exists(marker_file):  # pragma: no branch
            initial_size = os.path.getsize(marker_file)
            await anyio.sleep(0.3)
            size_after_wait = os.path.getsize(marker_file)
>           assert size_after_wait > initial_size, "Child process should be writing"
E           AssertionError: Child process should be writing
E           assert 0 > 0

tests/client/test_stdio.py:310: AssertionError
```

```python
________ TestChildProcessCleanup.test_nested_process_tree ________
[gw9] darwin -- Python 3.11.14

>           assert new_size > initial_size, f"{name} process should be writing"
E           AssertionError: parent process should be writing
E           assert 0 > 0

tests/client/test_stdio.py:413: AssertionError
```

```python
________ TestChildProcessCleanup.test_early_parent_exit ________
[gw9] darwin -- Python 3.11.14

>           assert size2 > size1, "Child should be writing"
E           AssertionError: Child should be writing
E           assert 0 > 0

tests/client/test_stdio.py:494: AssertionError
```

### Python & MCP Python SDK

```Text
Python 3.11.14, macOS (darwin)
Latest SDK version
```

<sub>[AI Disclaimer](https://gist.github.com/maxisbey/6123d132484e4c533eab519a2800693d)</sub>

--- a/src/mcp/os/win32/utilities.py
+++ b/src/mcp/os/win32/utilities.py
@@ -123,6 +123,11 @@ class FallbackProcess:
         """Return the process ID."""
         return self.popen.pid
 
+    @property
+    def returncode(self) -> int | None:
+        """Return the exit code, or ``None`` if the process has not yet terminated."""
+        return self.popen.returncode
+
 
 # ------------------------
 # Updated function

--- a/tests/client/test_stdio.py
+++ b/tests/client/test_stdio.py
@@ -335,7 +335,7 @@ async def _terminate_and_reap(proc: anyio.abc.Process | FallbackProcess) -> None
     safety net. Bounded by ``move_on_after`` to prevent hangs.
     """
     with anyio.move_on_after(5.0):
-        if getattr(proc, "returncode", None) is None:
+        if proc.returncode is None:
             await _terminate_process_tree(proc)
         await proc.wait()
         assert proc.stdin is not None

